scripts/build_docker.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# This script builds a Docker image of Coder containing the given binary, for
+# the given architecture. Only linux binaries are supported at this time.
+#
+# Usage: ./build_docker.sh --arch amd64 [--version 1.2.3] [--push]
+#
+# The --arch parameter is required and accepts a Golang arch specification. It
+# will be automatically mapped to a suitable architecture that Docker accepts
+# before being passed to `docker buildx build`.
+#
+# The image will be built and tagged against the image tag returned by
+# ./image_tag.sh.
+#
+# If no version is specified, defaults to the version from ./version.sh.
+#
+# If the --push parameter is supplied, the image will be pushed.
+#
+# Prints the image tag on success.
+
+set -euo pipefail
+# shellcheck source=scripts/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")/lib.sh"
+
+arch=""
+version=""
+push=0
+
+args="$(getopt -o "" -l arch:,version:,push -- "$@")"
+eval set -- "$args"
+while true; do
+	case "$1" in
+	--arch)
+		arch="$2"
+		shift 2
+		;;
+	--version)
+		version="$2"
+		shift 2
+		;;
+	--push)
+		push=1
+		shift
+		;;
+	--)
+		shift
+		break
+		;;
+	*)
+		error "Unrecognized option: $1"
+		;;
+	esac
+done
+
+if [[ "$arch" == "" ]]; then
+	error "The --arch parameter is required"
+fi
```

Comment:
On MacOS:

```shell
$ /opt/homebrew/bin/bash --version | head -1
GNU bash, version 5.1.16(1)-release (aarch64-apple-darwin21.1.0)
$ /opt/homebrew/bin/bash -x scripts/build_docker.sh --arch amd64 --version=0.0.0+testing
[...]
+ arch=
+ version=
+ push=0
++ getopt -o '' -l arch:,version:,push -- --arch amd64 --version=0.0.0+testing
+ args=' --  -l arch:,version:,push -- --arch amd64 --version=0.0.0+testing'
+ eval set -- ' --  -l arch:,version:,push -- --arch amd64 --version=0.0.0+testing'
++ set -- -- -l arch:,version:,push -- --arch amd64 --version=0.0.0+testing
+ true
+ case "$1" in
+ shift
+ break
+ [[ '' == '' ]]
+ error 'The --arch parameter is required'
+ log 'ERROR: The --arch parameter is required'
+ echo 'ERROR: The --arch parameter is required'
ERROR: The --arch parameter is required
+ exit 1
```

Reply:
This is caused by using BSD getopt instead of GNU getopt. The release pipeline installs the correct versions, and since these aren't called by users directly at the moment (and only by CI) I don't think we need to add version checking for now. Maybe in the future we can verify the version of bash, getopt and make.

Reply:
"Remember kids, Linux Is Not UniX"

Reply:
Actually, I'm stupid. Users will be calling this because of the makefile changes to use this :/

Reply:
Added some version checks for `bash`, `make` and `getopt`